uk_geo_utils/base_updater.py

```diff
@@ -0,0 +1,377 @@
+        self.stdout.write(f"Executing: {primary_key_rename_statement}")
+        self.cursor.execute(primary_key_rename_statement)
+
+        index_rename_statements = []
+        for index in self.indexes:
+            if self.index_exists(index["temp_index_name"]):
+                index_rename_statements.append(
+                    index["rename_temp_index_statement"]
+                )
+
+        for statement in index_rename_statements:
+            self.stdout.write(f"Executing: {statement}")
+            self.cursor.execute(statement)
+
+    def get_constraints_and_index_statements(self):
+        self.stdout.write(
+            f"Getting constraints and indexes for {self.table_name}"
+        )
+        self.primary_key_constraint = self.get_primary_key_constraint()
+        self.indexes = self.get_index_statements()
+        self.foreign_key_constraints = self.get_foreign_key_constraints()
+
+    def handle(self, **options):
+        if not check_memory():
+            raise Exception(
+                "This instance has less than the recommended memory. Try running the import from a larger instance."
+            )
+
+        self.get_data_path(options)
+
+        self.get_constraints_and_index_statements()
+
+        try:
+            # Create empty temp tables
+            self.create_temp_table()
+
+            # import data into the temp table
+            self.import_data_to_temp_table()
+
+            # Add temp primary keys
+            self.add_temp_primary_key()
+
+            # Add temp indexes
+            self.build_temp_indexes()
+
+            with transaction.atomic():
+                # Drop Foreign keys
+                if self.foreign_key_constraints:
+                    self.drop_foreign_keys()
+
+                # Rename old tables, pkeys and indexes (add _old)
+                self.rename_old_table()
+
+                # Rename temp table to original names, pkey and indexes
+                self.rename_temp_table()
+
+                # Add Foreign keys
+                if self.foreign_key_constraints:
+                    self.add_foreign_keys()
+
+            # Validate foreign key constraints
+            if self.foreign_key_constraints:
+                self.validate_foreign_keys()
+
+        finally:
+            self.db_cleanup()
+            self.file_cleanup()
+
+        self.stdout.write("...done")
+
+    def db_cleanup(self):
+        self.stdout.write("Dropping old table if exists...")
+        self.cursor.execute(
+            f"DROP TABLE IF EXISTS {self.table_name}_old CASCADE;"
+        )
+        self.stdout.write("Dropping temp table if exists...")
+        self.cursor.execute(
+            f"DROP TABLE IF EXISTS {self.temp_table_name} CASCADE;"
+        )
+
+    def file_cleanup(self):
+        if self.tempdir:
+            try:
+                shutil.rmtree(self.tempdir)
```

Comment:
Add a `self.stdout` logging line.

Reply:
is this a TODO?

Reply:
Yep. Just had a read through after pushing and figured that would be useful.

Reply:
1695cae4c63373